doc/source/how_to/execution/creating_custom_executors.ipynb

```diff
@@ -0,0 +1,481 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How to create a custom executor\n",
+    "\n",
+    "<font size=\"3\">\n",
+    "Each electron can utilise different so-called executors. These exectors define the low-level directions for the computation. They can specify different capabilities, _eg_, different hardware, different computation strategy, different logic, or simply differnt goals.\n",
+    "\n",
+    "Executors are plugins; any executor-plugins which are found are imported as classes in the covalent.executor name-space.\n",
+    "See the how-to on [choosing an executor to be used in an electron](choosing_executors.ipynb) for details on simply choosing an executor. Here, we go through the steps, line-by-line, on how to make your own.\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "There is a certain amount of boiler-plate Covalent code and imports that are required:\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# All executor plugins inherit from the BaseExecutor base class.\n",
+    "from covalent.executor import BaseExecutor\n",
+    "\n",
+    "# DispatchInfo objects are used to share info of a dispatched computation between different\n",
+    "# tasks (electrons) of the workflow (lattice).\n",
+    "from covalent._shared_files.util_classes import DispatchInfo\n",
+    "\n",
+    "# The function to be computed is in the form of a Covalent TransportableObject.\n",
+    "# This import is not strictly necessary, as it is only used for type-hints.\n",
+    "from covalent._workflow.transport import TransportableObject\n",
+    "\n",
+    "# The covalent logger module is a simple wrapper around the\n",
+    "# standard Python logging module.\n",
+    "from covalent._shared_files import logger\n",
+    "app_log = logger.app_log\n",
+    "log_stack_info = logger.log_stack_info\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<font size=\"3\">\n",
+    "We also need a few required standard Python imports:\n",
+    "</font>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```python\n",
+    "# Results of operations take the form of io.StringIO objects.\n",
+    "# These modules are used to capture those results.\n",
+    "import io\n",
```

Comment:
Here, `io.StringIO` objects will contain the print statement strings done inside an electron definition.

Reply:
I've re-worded this now, to make clear that the redirect is for.